kubernetes/structures.go

```diff
@@ -192,6 +218,18 @@ func isInternalKey(annotationKey string) bool {
 	return false
 }
 
+// This function verifies if a given Kubernetes metadata(annotations and labels) key
+// is among the keys that need to be ignored
+func isIgnoteKey(kubernetesMetadataKey string, ignoreKubernetesMetadata []string) bool {
+	for _, im := range ignoreKubernetesMetadata {
+		if ok, _ := regexp.MatchString(im, kubernetesMetadataKey); ok {
```

Comment:
We are suppressing the error here, but it is actually important for us here to capture if the regular expression failed to parse. If the user supplies an invalid regex then this is going to fail silently.

Reply:
Great catch, @jrhouston!

I have added provider level schema validation `validation.StringIsValidRegExp` for both new attributes. In that case, the practitioners will get an error as early as possible if they use invalid regular expressions. Since `validation.StringIsValidRegExp` and `regexp.MatchString` use the same RegExp validation mechanism(they simply compile the RegExp and throw an error if it is incorrect), we are safe to ignore errors at this stage.

I have also addressed other comments. 👍 

Thank you! 😊